### PR TITLE
[stress-tests] add the large_network_forming stress test

### PR DESCRIPTION
**Pull request by simonlingoogle**
This PR adds the `large_network_forming` stress test. 

The stress test creates 8x8 nodes and wait for them to form partitions. 
- Random packet loss is used to make the network not that stable so that corner cases can be checked. 
- The test executes for  10 hours in simulating time (much less in actual time).
- The test passes as long as no node crashes. 

This is one of the tests of #11, but I plan to divide it into multiple PRs to ease code reviewing.  

**This very stress test revealed subtle bugs fixed by https://github.com/openthread/openthread/pull/4860.**

### CODE REVIEW

File: pylibs/stress_tests/BaseStressTest.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
```
Comment (wgtdkp) [inline pylibs/stress_tests/BaseStressTest.py:26]:
```suggestion
# POSSIBILITY OF SUCH DAMAGE.
#
```

File: pylibs/stress_tests/StressTestResult.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    @property
+    def column_num(self):
+        return len(self.headers)
+
+    def append_row(self, *values: Any):
+        assert len(values) == len(self.headers)
+        self.rows.append(values)
+
+    def format(self):
```
Comment (wgtdkp) [inline pylibs/stress_tests/StressTestResult.py:76]: Can you give an example at here how the format will look like?

File: pylibs/stress_tests/large_network_forming.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+XGAP = 100
+YGAP = 100
+RADIO_RANGE = int(XGAP * 1.5)
+
+LARGE_N = 8
+PACKET_LOSS_RATIO = 0.9
+
+SIMULATE_TIME = 3600
+REPEAT = 10 if os.getenv('GITHUB_ACTIONS') else 1
```
Comment (wgtdkp) [inline pylibs/stress_tests/large_network_forming.py:40]: I think we should not depend on the platform specific env `GITHUB_ACTIONS`.

File: pylibs/stress_tests/StressTestResult.py (hunk 2 above)
Comment (simonlingoogle) [reply pylibs/stress_tests/StressTestResult.py:76]: The output is markdown formatted like this:

**[OTNS](https://github.com/openthread/ot-ns) Stress Tests Report Generated at 07/16 14:55:33**
#### Large Network Formation Test **Passed** _(07/16 06:53:51 ~ 07/16 06:55:33)_
| Simulation Time | Execution Time | Average Partition Count in 60s |
| :-: | :-: | :-: |
| 3600s | 101s | 15 |

File: pylibs/stress_tests/large_network_forming.py (hunk 3 above)
Comment (simonlingoogle) [reply pylibs/stress_tests/large_network_forming.py]: Changed to use `STRESS_LEVEL`. 
Each test is responsible for running about `STRESS_LEVEL` minutes. 
For example, if `STRESS_LEVEL=10`, the script itself should run for 10 minutes. It's not a hard limit, just some suggestion for each script.

**Comment by simonlingoogle** [comment]
Further comments? @wgtdkp
